internal/dag/dag.go

```diff
@@ -278,6 +265,8 @@ type Cluster struct {
 	// The load balancer type to use when picking a host in the cluster.
 	// See https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/cds.proto#envoy-api-enum-cluster-lbpolicy
 	LoadBalancerStrategy string
+
+	HealthCheck *ingressroutev1.HealthCheck
```

Comment:
Should we also not tie the IngressRoute struct to the dag?

Reply:
Yes, I want to redo this type into something like HealthCheckPolicy. I do want to remove the depdendency on the ingressroute type, but this PR is just moving the field. I'll do the rest in a followup

Reply:
ok sounds good